File: bindings/android/src/test/java/org/indunet/robot/bus/RobotBusAndroidTest.java
package org.indunet.robot.bus;

import static org.junit.Assert.assertTrue;
import static org.junit.Assert.fail;

import org.junit.Test;

/**
 * Android-module smoke tests. Full gRPC / typed Node coverage lives in the JVM
 * artifact ({@code bindings/java} {@code GrpcNodeTest} / {@code TypedApiTest});
 * this AAR only adds {@link RobotBusAndroid#init} + jniLibs.
 */
public class RobotBusAndroidTest {
    @Test
    public void initFailsWithoutJniLibsOnHostJvm() {
        try {
            // Context is unused today; host JVM has no librobot_bus_c in
            // java.library.path the way APK jniLibs packaging provides it.
            RobotBusAndroid.init(null);
            fail("expected UnsatisfiedLinkError without Android jniLibs");
        } catch (UnsatisfiedLinkError expected) {
            assertTrue(expected.getMessage() == null || expected.getMessage().contains("robot_bus_c"));
        }
    }
}

File: bindings/android/src/test/kotlin/org/indunet/robot/bus/RobotBusAndroidTest.kt
package org.indunet.robot.bus

import org.junit.Assert.assertTrue
import org.junit.Assert.fail
import org.junit.Test

/**
 * Android entry smoke tests. Parameter / typed API coverage uses desktop
 * `librobot_bus_c` via `ROBOT_BUS_NATIVE_DIR` (see [NodeParametersTest], [TypedApiTest]).
 */
class RobotBusAndroidTest {
    @Test
    fun initFailsWithoutJniLibsOnHostJvm() {
        try {
            // Context is unused today; host JVM has no librobot_bus_c in
            // java.library.path the way APK jniLibs packaging provides it.
            RobotBusAndroid.init(null)
            fail("expected UnsatisfiedLinkError without Android jniLibs")
        } catch (expected: UnsatisfiedLinkError) {
            assertTrue(
                expected.message == null || expected.message!!.contains("robot_bus_c"),
            )
        }

        // Failure must not mark initialized; a second call should still fail the same way.
        try {
            RobotBusAndroid.init(null)
            fail("expected UnsatisfiedLinkError on retry after failed init")
        } catch (expected: UnsatisfiedLinkError) {
            assertTrue(
                expected.message == null || expected.message!!.contains("robot_bus_c"),
            )
        }
    }
}
